doc/classes/Variant.xml

```diff
@@ -1,49 +1,63 @@
+		var boo = "boo is a string!";
 		var ref = new RefCounted(); // var is especially useful when used together with a constructor.
 
 		// Godot also provides a Variant type that works like a union of all the Variant-compatible types.
 		Variant fooVar = 2; // fooVar is dynamically an integer (stored as a `long` in the Variant type).
 		fooVar = "Now fooVar is a string!";
-		fooVar = new RefCounted(); // fooVar is a GodotObject.
+		fooVar = new RefCounted(); // fooVar is now a GodotObject.
 		[/csharp]
 		[/codeblocks]
-		Godot tracks all scripting API variables within Variants. Without even realizing it, you use Variants all the time. When a particular language enforces its own rules for keeping data typed, then that language is applying its own custom logic over the base Variant scripting API.
-		- GDScript automatically wrap values in them. It keeps all data in plain Variants by default and then optionally enforces custom static typing rules on variable types.
-		- C# is statically typed, but uses its own implementation of the Variant type in place of Godot's [Variant] class when it needs to represent a dynamic value. C# Variant can be assigned any compatible type implicitly but converting requires an explicit cast.
-		The global [method @GlobalScope.typeof] function returns the enumerated value of the Variant type stored in the current variable (see [enum Variant.Type]).
+
+		Godot tracks all scripting API variables within Variants. When a particular language enforces its own rules for keeping data typed, that language is applying its own custom logic over the base Variant scripting API:
+		- GDScript automatically wraps values in them. It keeps all data in plain Variants by default and then optionally enforces custom static typing rules on variable types.
+		- C# is statically typed, but uses its own implementation of Variant when it needs to represent a dynamic value. A [Variant] can be assigned any compatible type implicitly, but converting requires an explicit cast.
+
+		A Variant in Godot takes up only 20 bytes and can store almost any datatype inside of it. Those can be roughly categorized as:
+		- The [code]null[/code] type. It represents a Variant with no value assigned and is commonly returned by functions when a result cannot be found.
+		- Atomic types, the most basic data types in Godot: [bool], [int], [float], [String]
+		- Complex types, which consist of atomic types. They represent common constructs like vectors and colors, and allow you to easily work with them: [Color], [Vector2], [Transform3D], etc.
+		- Container types, which can store multiple other Variants, including other containers: [Dictionary], [Array]. This also includes packed arrays, which are like an [Array] that can only hold one type of data and handle it very efficiently: [PackedColorArray], [PackedInt32Array], etc.
+		- The [Object] type, inherited by every object in Godot, such as [Node], [Resource], [EditorFileSystemImportFormatSupportQuery], etc.
```

Comment:
"EditorFileSystemImportFormatSupportQuery" was a dumb joke I forgot to remove.

Reply:
Should have been Marker2D for the joke as the contender for the most useless node